### PR TITLE
Optimization for UCLA reports

### PR DESCRIPTION
**Pull request by lwyszomi**
@czue @snopoke

### CODE REVIEW

File: custom/succeed/reports/patient_submissions.py [hunk 1]
```diff
@@ -1,132 +1,109 @@
...
+from sqlagg.columns import SimpleColumn
+from sqlagg.filters import EQ
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
-from corehq.apps.reports.filters.search import SearchFilter
-from custom.succeed.reports import SUBMISSION_SELECT_FIELDS, EMPTY_FIELD, INTERACTION_OUTPUT_DATE_FORMAT
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.reports.sqlreport import SqlData, DatabaseColumn
+from corehq.apps.reports.standard import CustomProjectReport, ProjectReportParametersMixin
+from corehq.apps.userreports.sql import get_table_name
+from custom.succeed.reports import EMPTY_FIELD, INTERACTION_OUTPUT_DATE_FORMAT
```
Comment (dimagimon) [inline custom/succeed/reports/patient_submissions.py]: F401 'EMPTY_FIELD' imported but unused

Comment (dimagimon) [inline custom/succeed/reports/patient_submissions.py]: F401 'INTERACTION_OUTPUT_DATE_FORMAT' imported but unused

**Comment by snopoke** [comment]
Looks good but I'm curious to know why you decided to switch from using ES to using a custom data source?

**Comment by lwyszomi** [comment]
@snopoke 

Because data source works quicker.

File: custom/succeed/models.py [hunk 2]
```diff
@@ -2,5 +2,106 @@
...
+from fluff.filters import CustomFilter
 
-class _(Document): pass
\ No newline at end of file
+LAST_VISIT_DICT = {
+    'name': 'last',
+    'days': -1
+}
+
+
```
Comment (dimagimon) [inline custom/succeed/models.py]: E701 multiple statements on one line (colon)

File: custom/succeed/models.py [hunk 3]
```diff
@@ -2,5 +2,106 @@
...
+    bp_category = flat_field(lambda case: get_property(case, 'BP_category'))
+    care_site = flat_field(lambda case: get_property(case, 'care_site').lower())
+    is_active = flat_field(lambda case: is_active(case))
+    visit_name = flat_field(lambda case: visit_name(case))
+    visit_days = flat_field(lambda case: visit_days(case))
+    last_interaction = flat_field(lambda case: last_interaction(case))
+
+    emitter = RandomizationDate()
+
```
Comment (dimagimon) [inline custom/succeed/models.py]: W292 no newline at end of file

File: custom/succeed/reports/all_patients.py [hunk 4]
```diff
@@ -1,174 +1,113 @@
...
-from corehq.apps.reports.standard.cases.data_sources import CaseDisplay
-from corehq.elastic import es_query
-from corehq.pillows.base import restore_property_dict
+from corehq.apps.reports.standard import CustomProjectReport, ProjectReportParametersMixin
 from django.utils import html
-import dateutil
-from corehq.pillows.mappings.reportcase_mapping import REPORT_CASE_INDEX
-from custom.succeed.reports import VISIT_SCHEDULE, LAST_INTERACTION_LIST, EMPTY_FIELD, CM7, PM3,\
-    CM_APP_CM_MODULE, OUTPUT_DATE_FORMAT, INPUT_DATE_FORMAT
+from custom.succeed.reports import  EMPTY_FIELD, CM7, \
```
Comment (dimagimon) [inline custom/succeed/reports/all_patients.py]: E271 multiple spaces after keyword

File: custom/succeed/models.py [hunk 5]
```diff
@@ -2,5 +2,104 @@
...
 from . import signals
+from corehq.apps.users.models import CommCareCase
+from custom.succeed.reports import VISIT_SCHEDULE, LAST_INTERACTION_LIST, PM3
+import fluff
+from custom.utils.utils import flat_field
+from fluff.filters import CustomFilter
 
-class _(Document): pass
\ No newline at end of file
+
```
Comment (dimagimon) [inline custom/succeed/models.py]: E701 multiple statements on one line (colon)

**Comment by lwyszomi** [comment]
@czue @snopoke 

I updated the PR, added second part of optimization for UCLA reports. Please merge if all looks good.

need reindex new fluff

**Comment by czue** [comment]
I'll let @snopoke review/merge. I believe that it should auto-reindex, but am not 100% sure about that.

File: custom/succeed/reports/all_patients.py [hunk 6]
```diff
@@ -1,174 +1,112 @@
...
-            PatientTaskListReport.get_url(*[self.case["domain"]]) +
-            "?patient_id=%s&task_status=open" % self.case["_id"])
-        if url:
-            return html.mark_safe("<a class='ajax_dialog' href='%s' target='_blank'>Tasks</a>" % url)
-        else:
-            return "%s (bad ID format)" % self.case_name
 
+def test(val):
+    print val
+    return val
```
Comment (snopoke) [inline custom/succeed/reports/all_patients.py]: I think this can go

**Comment by snopoke** [comment]
A bit difficult to read the diff but looks OK. Merging but see comments.

Shouldn't be any issue with indexing.

**Pull request merged**